analytic_engine/src/instance/flush_compaction.rs

```diff
@@ -167,6 +167,19 @@ pub struct TableFlushRequest {
     pub table_data: TableDataRef,
     /// Max id of memtable to flush (inclusive).
     pub max_memtable_id: MemTableId,
+    /// Max sequence number to flush (inclusive).
+    pub max_sequence: SequenceNumber,
+    /// Flush policy.
+    pub policy: TableFlushPolicy,
+}
+
+/// Policy of how to perform flush operation.
+#[derive(Debug)]
+pub enum TableFlushPolicy {
+    /// Dump memtable to sst file.
+    Dump,
+    /// Drop memtables.
+    Purge,
```

Comment:
I guess we need one policy called `Block` which handles that the follower's memtables can't be flushed until the flushed sequence number from its leader is greater than its memtable to flush.

Reply:
This is covered in `pick_memtables_to_flush`. It only picks memtables whose `last_sequence` ≤ sequence number in the flush request.